controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -795,6 +798,9 @@ func (pp *portPublisher) endpointsToAddresses(endpoints *corev1.Endpoints) Addre
 				address, id := pp.newServiceRefAddress(resolvedPort, endpoint.IP, endpoints.Name, endpoints.Namespace)
 				address.Identity, address.AuthorityOverride = identity, authorityOverride
 
+				if _, ok := addresses[id]; ok && resolvedPort == undefinedEndpointPort {
+					continue
+				}
```

Comment:
This can never be true given the addition above.

Reply:
Thanks, had forgotten to remove the old mitigation 🤦, fixed :-)